build.gradle

```diff
@@ -76,26 +77,27 @@ for (String line : versionLines) {
     int major = Integer.parseInt(match.group(1))
     int minor = Integer.parseInt(match.group(2))
     int bugfix = Integer.parseInt(match.group(3))
-    String versionStr = "${major}.${minor}.${bugfix}"
-    if (currentVersion != versionStr) {
-      versions.add(versionStr)
+    Version foundVersion = new Version(major, minor, bugfix, false)
+    if (currentVersion != foundVersion) {
+      versions.add(foundVersion)
     }
     if (major == prevMajor && minor > lastPrevMinor) {
       prevMinorIndex = versions.size() - 1
       lastPrevMinor = minor
     }
   }
 }
-if (versions.toSorted() != versions) {
-  throw new GradleException('Versions.java contains out of order version constants')
+if (versions.toSorted { it.id } != versions) {
+  println "Versions: ${versions}" 
```

Comment:
No, this is meant to print the versions before failing if we find the versions in Version.java are not in sorted order.